src/trace.ml

```diff
@@ -589,31 +620,53 @@ module Make_commands (Backend : Backend_intf.S) = struct
            | Some path -> path
            | None -> failwithf "Can't find executable for %s" prog ()
          in
-         record_opt_fn ~executable ~f:(fun opts ->
-             let elf = Elf.create opts.executable in
-             let%bind range_symbols =
-               evaluate_trace_filter ~trace_filter:opts.trace_filter ~elf
-             in
-             let%bind pid =
-               let argv = prog :: List.concat (Option.to_list argv) in
-               run_and_record
-                 opts
+         let%bind () =
+           record_opt_fn ~executable ~f:(fun opts ->
+               let elf = Elf.create opts.executable in
+               let%bind range_symbols =
+                 evaluate_trace_filter ~trace_filter:opts.trace_filter ~elf
+               in
+               let%bind pid =
+                 let argv = prog :: List.concat (Option.to_list argv) in
+                 run_and_record
+                   opts
+                   ~elf
+                   ~debug_print_perf_commands
+                   ~prog
+                   ~argv
+                   ~collection_mode:opts.collection_mode
+               in
+               let%bind.Deferred perf_maps = Perf_map.Table.load_by_pids [ pid ] in
+               decode_to_trace
+                 ~perf_maps
+                 ?range_symbols
                  ~elf
+                 ~trace_scope:opts.trace_scope
                  ~debug_print_perf_commands
-                 ~prog
-                 ~argv
+                 ~record_dir:opts.record_dir
                  ~collection_mode:opts.collection_mode
-             in
-             let%bind.Deferred perf_maps = Perf_map.Table.load_by_pids [ pid ] in
-             decode_to_trace
-               ~perf_maps
-               ?range_symbols
-               ~elf
-               ~trace_scope:opts.trace_scope
-               ~debug_print_perf_commands
-               ~record_dir:opts.record_dir
-               ~collection_mode:opts.collection_mode
-               decode_opts))
+                 decode_opts)
+         in
+         let%bind.Deferred () =
+           match url with
+           | None -> Deferred.return ()
+           | Some url -> Async_shell.run "open" [ url ]
+         in
+         let output_config = decode_opts.Decode_opts.output_config in
+         let output = Tracing_tool_output.output output_config in
+         let output_file =
+           match output with
+           | `Sexp _ -> failwith "unimplemented"
+           | `Fuchsia store_path -> store_path
+         in
+         let%bind.Deferred () =
+           match trace_processor_exe with
+           | None ->
+             print_endline "warning: must use local processor on large trace files";
+             Deferred.return ()
+           | Some processor_path -> Async_shell.run processor_path [ "-D"; output_file ]
```

Comment:
Can we use something like https://github.com/janestreet/magic-trace/blob/872c648000ecff99c2035d239df4a53a8c5f91a0/src/perf_tool_backend.ml#L70 here, to avoid the trace processor shell outliving magic-trace, in case magic-trace is <kbd>Ctrl</kbd>+<kbd>C</kbd>'d. May want to extract it into a helper module like `process.ml` or something.